test/e2e/es/license_test.go

```diff
@@ -127,3 +129,61 @@ func TestEnterpriseTrialLicense(t *testing.T) {
 
 	test.Sequence(initStepsFn, stepsFn, esBuilder).RunSequential(t)
 }
+
+// TestEnterpriseTrialExtension tests that trial extensions can be successfully applied and take effect.
+func TestEnterpriseTrialExtension(t *testing.T) {
```

Comment:
Can you add some comments describing what this test is doing? (it manipulates both ECK & ES licenses IIUC, which is rather complicated)